src/config.rs

```diff
@@ -2234,8 +2234,9 @@ pub struct BackupConfig {
 
 impl BackupConfig {
     pub fn validate(&self) -> Result<(), Box<dyn Error>> {
-        if self.num_threads == 0 {
-            return Err("backup.num_threads cannot be 0".into());
+        let limit = SysQuota::cpu_cores_quota() as usize;
+        if self.num_threads == 0 || self.num_threads > limit {
```

Comment:
Same concern as https://github.com/tikv/tikv/pull/12068#discussion_r825657740, also the actual number of threads used by backup is decided by the backup request, `num_threads` is used by a token limiter

Reply:
I discussed it with @YuJuncen before, since backup is cpu intensive, so it doesn't make much sense to set the number of threads above 100% cpu quota.  And it has the feature of automatic current limiting when the load is high.  /cc @YuJuncen

Reply:
@NingLin-P 
> actual number of threads used by backup is decided by the backup request.
The number of threads isn't decided by the request more because the risk of misconfigure leading to panicking. (see https://github.com/tikv/tikv/pull/8193)

The step of saving backup files into external storage is now asynchronous, which means those threads would only be blocked by scanning the local MVCC(which is fast enough and hardly blocking, in some test cases, with the NVME disks,  CPU usage of those threads become up to 95%.). Most of CPUs were used for doing `zstd` compressing. So IMO, we cannot take much good from backing up with a thread pool huger than the CPU number size.